### PR TITLE
Range: add pinFormatter and `move` event emitter

### LINKED ISSUE #3295: [BUG] ion-range is missing ionChange when knob is moving
- [ ] I have written a [good issue](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Issue) 

## Describe the bug
When using Kirby range, the knob only emits an onChange when the knob is released, and not when the knob is moved. 

## Describe how to reproduce the bug

When the knob is moved through the range, it should emit a change every time it hits a designated point, so other components can be updated. 

## Are there any additional context?

Maybe forward the pinFormatter property through kirby-range?
<hr />

## Checklist:

The following tasks should be carried out in sequence in order to follow [the process of contributing](https://github.com/kirbydesign/designsystem/blob/main/.github/CONTRIBUTING.md/#the-process-of-contributing) correctly.

### Verification
To make sure the bug is not intended behaviour; it should be verified by a member of team Kirby before moving on to implementation. 

- [ ] Make sure the `NOT verified` label has been removed by a member of [team Kirby](https://github.com/kirbydesign/designsystem/blob/main/.github/SUPPORT.md#team-kirby); do not proceed until this is done

### Implementation 
The contributor who wants to implement this issue should: 

- [ ] Make sure you have read: "[Before you get coding](../CONTRIBUTING.md/#before-you-get-coding)".
- [ ] Signal to others you are working on the issue by assigning yourself.
- [ ] Create a branch from the [develop branch](https://github.com/kirbydesign/designsystem/tree/develop) following our [branch naming convention](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Branch). 
- [ ] Create a test that reproduces the bug following guidelines in: "[The good: Test](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Test)". 
- [ ] Write code to fix the bug

### Review
Once the issue has been implemented and is ready for review:

- [ ] Do a [self-review](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Self-review). 
- [ ] Create a pull-request.

### LINKED ISSUE #3296: [Enhancement] Add pinFormatter function to kirby-range
- [x] I have written a [good issue](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Issue) 

## Describe the enhancement
In Ionic 7 the range component no longer emits the `ionChange` event for each value change, but instead only when the user releases the knob after dragging or when the user moves the knob with keyboard arrows. 

The change event is proxied directly by us, and thus the same is true for kirby-range in Kirby v9. Some consumers use the existing value change functionality for updating their view with a 'currently selected' value when the user drags along the knob.  

## Describe the solution you'd like
We can provide an alternative to the above functionality by exposing the `pinFormatter` functionality (see [here](https://ionicframework.com/docs/api/range#pinformatter)) of ion-range. This enables teams to format the value  shown in the pin in a user-friendly manner: 

image

Alternatively the original functionality can be achieved by changing the behavior of our own change event to proxying the `ionInput` event instead, that fires on every change.

Seeing as the breaking change from Ionic is a performance optimization, because the range can fire _a lot!_ of events when dragged, I think we should adopt Ionics `ionChange` behavior and use the `pinFormatter` for these types of use-cases.

<hr />

## Checklist:

The following tasks should be carried out in sequence in order to follow [the process of contributing](https://github.com/kirbydesign/designsystem/blob/main/.github/CONTRIBUTING.md/#the-process-of-contributing) correctly.

### Refinement

- [ ] Request that the issue is [UX refined](https://github.com/kirbydesign/designsystem/blob/main/.github/CONTRIBUTING.md/#ux-refinement); do not proceed until this is done.
- [ ] Request that the issue is [tech refined](https://github.com/kirbydesign/designsystem/blob/main/.github/CONTRIBUTING.md/#tech-refinement); do not proceed until this is done.

### Implementation 
The contributor who wants to implement this issue should: 

- [ ] Make sure you have read: "[Before you get coding](../CONTRIBUTING.md/#before-you-get-coding)".
- [ ] Signal to others you are working on the issue by assigning yourself.
- [ ] Create a branch from the [develop branch](https://github.com/kirbydesign/designsystem/tree/develop) following our [branch naming convention](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Branch). 
- [ ] Publish a WIP implementation to Github as a draft PR and ask for feedback. 
- [ ] Make sure you have implemented tests following the guidelines in: "[The good: Test](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Test)".
- [ ] Update the [cookbook](https://cookbook.kirby.design) with examples and showcases.

### Review
Once the issue has been implemented and is ready for review:

- [ ] Do a [self-review](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Self-review). 
- [ ] Create a pull-request. If you created a draft PR during implementation you can just mark that as "ready for review".

### PR DESCRIPTION
**Pull request by RasmusKjeldgaard**
## Which issue does this PR close?

This PR closes #3296 closes #3295

## What is the new behavior?
See issues. 

TL;DR: To preserve Ionics performance enhancements to the range's change event, but still allow applications to react on each value change if they really need it, the `ionInput` event is now proxied by range as `move` so applications can get notified whenever the knob is moved to a new value.

At the same time we also add the `pinFormatter` function as another, and in most cases better, migration path for use-cases that need to show the currently "selected" value in the range before the user lets go of the knob.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Checklist:

The following tasks should be carried out in sequence in order to follow [the process of contributing](https://github.com/kirbydesign/designsystem/blob/main/.github/CONTRIBUTING.md/#the-process-of-contributing) correctly.

### Reminders
- [ ] Make sure you have implemented tests following the guidelines in: "[The good: Test](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Test)".
- [ ] Make sure you have updated the cookbook with examples and showcases (for bug fixes, enhancements & new components).

### Review  
- [ ] Determine if your changes are a fix, feature or breaking-change, and add the matching label to your PR. If it is tooling, dependency updates or similar, add ignore-for-release.
- [ ] Do a [self-review](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Self-review).
- [ ] Request that the changes are code-reviewed 
- [ ] Request that the changes are [UX reviewed](https://github.com/kirbydesign/designsystem/blob/main/.github/CONTRIBUTING.md/#ux-review) (only necessary if your PR introduces visual changes)

When the pull request has been approved it will be merged to `develop` by Team Kirby.

### CODE REVIEW

File: apps/cookbook/src/app/showcase/range-showcase/range-showcase.component.html [hunk 1]
```diff
@@ -25,6 +25,15 @@ <h2>Pin - the slider value is shown when the knob is pressed</h2>
     <cookbook-range-pin-example #pinExample></cookbook-range-pin-example>
   </cookbook-example-viewer>
 
+  <p>
+    Additionally, the
+    <code>pinFormatter</code>
+    function can be supplied to customize the formatting of the range value shown to the user in the
+    pin. It defaults to the range sliders integer-value.
```
Comment (jakobe) [inline apps/cookbook/src/app/showcase/range-showcase/range-showcase.component.html]:
```suggestion
    pin. It defaults to the range slider's integer-value.
```

File: apps/cookbook/src/app/showcase/range-showcase/range-showcase.component.ts [hunk 2]
```diff
@@ -65,7 +70,14 @@ export class RangeShowcaseComponent {
...
       name: 'change',
-      description: 'Emits events when the value changes.',
+      description:
+        'Emitted when the user modifies the value by releasing the knob or moving the knob with the keyboard',
+      signature: '() => EventEmitter<number>',
+    },
+    {
+      name: 'move',
+      description:
+        'Emitted for each distinct change whenever the knob is moved by the user. Unlike the change event it fires continuously while the user is dragging the knob.',
```
Comment (jakobe) [inline apps/cookbook/src/app/showcase/range-showcase/range-showcase.component.ts]:
```suggestion
        'Emitted for each distinct change whenever the knob is moved by the user. Unlike the `change` event it fires continuously while the user is dragging the knob.',
```

File: libs/designsystem/range/src/range.component.ts [hunk 3]
```diff
@@ -6,11 +6,13 @@ import {
...
   Input,
   OnChanges,
+  OnInit,
   Output,
   SimpleChanges,
+  ViewChild,
 } from '@angular/core';
 import { ControlValueAccessor, NG_VALUE_ACCESSOR } from '@angular/forms';
-import { IonicModule } from '@ionic/angular';
+import { IonicModule, IonRange } from '@ionic/angular';
```
Comment (jakobe) [inline libs/designsystem/range/src/range.component.ts:15]: Do we need both or should `IonicModule` be replaced by `IonRange`? 🤔

File: apps/cookbook/src/app/examples/range-example/examples/pin.component.ts [hunk 4]
```diff
@@ -4,10 +4,20 @@ const config = {
   selector: 'cookbook-range-pin-example',
   template: `<kirby-range pin="true" minLabel="Min value" maxLabel="Max value" max="15" min="1"></kirby-range>`,
 };
+
+const pinFormatterExample = {
+  template: `<kirby-range pin="true" [pinFormatter]="pinFormatter" minLabel="Min value" maxLabel="Max value" max="15" min="1"></kirby-range>`,
+  function: `pinFormatter(value: number) {
+    return \`\${value}%\`;
+  }`,
```
Comment (jakobe) [inline apps/cookbook/src/app/examples/range-example/examples/pin.component.ts]:
```suggestion
  function: `pinFormatter(value: number) {
  return \`\${value}%\`;
}`,
```

**Review by jakobe — CHANGES_REQUESTED** [review]

File: libs/designsystem/range/src/range.component.ts (hunk 3 above)
Comment (RasmusKjeldgaard) [reply libs/designsystem/range/src/range.component.ts:15]: We need IonicModule for `ion-range` to be recognizable in the template. And IonRange is not standalone so we cannot import it directly, so I think it has to be this way.

image

**Review by jakobe — APPROVED** [review]
Good work 👏🏻
